platform/osgi/platform-osgi-condpermadmin/src/main/java/org/codice/ddf/condpermadmin/PermissionActivator.java

```diff
@@ -0,0 +1,205 @@
+      buildConditionalPermissionInfo(
+          conditionalPermissionAdmin, denyEntries, denyInfos, ConditionalPermissionInfo.DENY);
+
+      Priority priority = parsedPolicy.getPriority();
+      allGrantInfos.addAll(grantInfos);
+      allDenyInfos.addAll(denyInfos);
+      if (priorityResult == null) {
+        priorityResult = priority;
+      } else if (priority != priorityResult) {
+        // if they don't match, then we can't make a determination on the priority, so we'll
+        // default to deny
+        priorityResult = Priority.DENY;
+      }
+    }
+
+    if (priorityResult == Priority.GRANT) {
+      conditionalPermissionUpdate.getConditionalPermissionInfos().addAll(allGrantInfos);
+      conditionalPermissionUpdate.getConditionalPermissionInfos().addAll(allDenyInfos);
+      conditionalPermissionUpdate
+          .getConditionalPermissionInfos()
+          .add(getAllPermission(conditionalPermissionAdmin, ConditionalPermissionInfo.ALLOW));
+    } else if (priorityResult == Priority.DENY) {
+      conditionalPermissionUpdate.getConditionalPermissionInfos().addAll(allDenyInfos);
+      conditionalPermissionUpdate.getConditionalPermissionInfos().addAll(allGrantInfos);
+      conditionalPermissionUpdate
+          .getConditionalPermissionInfos()
+          .add(getAllPermission(conditionalPermissionAdmin, ConditionalPermissionInfo.DENY));
+    }
+
+    conditionalPermissionUpdate.commit();
+  }
+
+  private ConditionalPermissionInfo getAllPermission(
+      ConditionalPermissionAdmin conditionalPermissionAdmin, String type) {
+    return conditionalPermissionAdmin.newConditionalPermissionInfo(
+        null,
+        null,
+        new PermissionInfo[] {new PermissionInfo("java.security.AllPermission", "", "")},
+        type);
+  }
+
+  private void buildConditionalPermissionInfo(
+      ConditionalPermissionAdmin conditionalPermissionAdmin,
+      List<ParsedPolicyEntry> entries,
+      List<ConditionalPermissionInfo> infos,
+      String type) {
+    for (ParsedPolicyEntry parsedPolicyEntry : entries) {
+      List<ParsedPermission> permissions = parsedPolicyEntry.getPermissions();
+      PermissionInfo[] permissionInfos = new PermissionInfo[permissions.size()];
+      int index = 0;
+      for (ParsedPermission parsedPermission : permissions) {
+        permissionInfos[index++] =
+            new PermissionInfo(
+                parsedPermission.getPermissionType(),
+                replaceSystemProperties(parsedPermission.getPermissionName()),
+                parsedPermission.getActions());
+      }
+      List<ConditionInfo> conditionInfos = new ArrayList<>();
+      addCodebase(parsedPolicyEntry, conditionInfos);
+      addSignedBy(parsedPolicyEntry, conditionInfos);
+      addPrincipals(parsedPolicyEntry, conditionInfos);
+      infos.add(
+          conditionalPermissionAdmin.newConditionalPermissionInfo(
+              null,
+              (conditionInfos.isEmpty()) ? null : conditionInfos.toArray(new ConditionInfo[0]),
```

Comment:
✏️ More optional to do new ConditionInfo[conditionInfos.size()] to avoid creating an array for nothing